geomstats/_backend/pytorch/__init__.py

```diff
@@ -92,8 +92,17 @@ def empty(shape, dtype=float64):
     return torch.empty(*shape, dtype=dtype)
 
 
-def split(ary, indices_or_sections, axis=0):
-    return torch.split(ary, indices_or_sections, dim=axis)
+def split(x, indices_or_sections, axis=0):
+    if isinstance(indices_or_sections, int):
+        indices_or_sections = indices_or_sections // x.shape[axis]
```

Comment:
shouldn't it be the inverse? `torch.split(x, k)` will return n // k (+1) arrays of size k (except the last one) while `numpy.split(x, k)` would return k arrays of size n // k, where `n = x.shape[axis]`, and throw an error if `n % k != 0`

Reply:
True, I had mixed them up, and everything was working fine since we were in a case were `indices_or_sections // x.shape[axis] = 1` ...